### PR TITLE
android app: API, add chat sheet and view with QR code

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt [hunk 1]
```diff
@@ -70,47 +64,122 @@ open class ChatController(val ctrl: ChatCtrl) {
...
 
   suspend fun apiSendMessage(type: ChatType, id: Long, mc: MsgContent): AChatItem? {
     val r = sendCmd(CC.ApiSendMessage(type, id, mc))
     if (r is CR.NewChatItem ) return r.chatItem
-    Log.d("SIMPLEX", "apiSendMessage bad response: ${r.toString()}")
+    Log.d("SIMPLEX", "apiSendMessage bad response: ${r.responseType} ${r.details}")
+    return null
+  }
+
+  suspend fun apiAddContact(): String? {
```
Comment (IanRDavies) [inline apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt:104]: Why does this not take any arguments? It feels like it should take some identifier of the contact to be added

File: apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt [hunk 2]
```diff
@@ -206,15 +273,67 @@ abstract class CC {
...
   }
 
   class ApiSendMessage(val type: ChatType, val id: Long, val mc: MsgContent): CC() {
-    override val cmdString get() = "/_send ${CC.chatRef(type, id)} ${mc.cmdString}"
+    override val cmdString get() = "/_send ${chatRef(type, id)} ${mc.cmdString}"
     override val cmdType get() = "apiGetChat"
   }
 
+  class AddContact: CC() {
+    override val cmdString get() = "/connect"
```
Comment (IanRDavies) [inline apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt:286]: Shouldn't this take an argument of the contact address to connect to?

File: apps/android/app/src/main/java/chat/simplex/app/views/helpers/SimpleButton.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  }
+}
+
+@Preview
+@Composable
+fun PreviewCloseSheetBar() {
+  SimpleXTheme {
+    SimpleButton(text = "Share", icon = Icons.Outlined.Share, click = {})
+  }
+}
```
Comment (IanRDavies) [inline apps/android/app/src/main/java/chat/simplex/app/views/helpers/SimpleButton.kt:44]:
```suggestion
}

```

File: apps/android/app/src/main/java/chat/simplex/app/MainActivity.kt [hunk 4]
```diff
@@ -5,6 +5,7 @@ import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.activity.viewModels
+import androidx.compose.material.ExperimentalMaterialApi
```
[the remark this answers is not in the document]
Comment (epoberezkin) [reply apps/android/app/src/main/java/chat/simplex/app/MainActivity.kt:8]: this is Compose experimental api that is needed for BottomSheetScaffold, it's not Android API.

File: apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt (hunk 1 above)
Comment (epoberezkin) [reply apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt:104]: At the time it's created you don't know it. We could consider giving names to this invitations, but now the name arrives in the profile when the invitation is accepted by the other party

Comment (epoberezkin) [reply apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt:104]: These invitations links will expire after 24-48 hours.

File: apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt (hunk 2 above)
Comment (epoberezkin) [reply apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt:286]: it's Connect that takes it - the counterpart to this

**Review by IanRDavies — APPROVED** [review]

**Pull request merged**